runtime/bcutil/ROMClassBuilder.cpp

```diff
@@ -291,7 +291,11 @@ ROMClassBuilder::handleAnonClassName(J9CfrClassFile *classfile, bool *isLambda,
 	 */
 	if (context->isClassHidden()
 	&& !context->isHiddenClassOptStrongSet()
+#if JAVA_SPEC_VERSION <= 17
 	&& !context->isHiddenClassOptNestmateSet()
+#else /* JAVA_SPEC_VERSION <= 17 */
+	&& context->isHiddenClassOptNestmateSet()
+#endif /* JAVA_SPEC_VERSION <= 17 */
```

Comment:
Please add a comment explaining why the condition needs to be inverted beginning with Java 18.

Reply:
A comment has been added.